src/js/tech/tech.js

```diff
@@ -851,13 +851,23 @@ Tech.withSourceHandlers = function(_Tech) {
     return this;
   };
 
+  let previousSource;
+
   // On the first loadstart after setSource
   _Tech.prototype.firstLoadStartListener_ = function() {
+    previousSource = this.currentSource_;
     this.one(this.el_, 'loadstart', _Tech.prototype.successiveLoadStartListener_);
   };
 
   // On successive loadstarts when setSource has not been called again
   _Tech.prototype.successiveLoadStartListener_ = function() {
+
+    // if the second loadstart is for the same source
+    // it is an invalid loadstart, set this event to trigger again
+    if (previousSource.src === this.currentSource_.src) {
```

Comment:
I think that there are two scenarios, the first of which was being checked for already:

Scenario 1:
1. setSource is called
2. `loadstart` happens (`previousSource` will be set to `this.currentSource_` for thes first time)
3. currentSource_ is changed
4. `loadstart` happens again (`previousSource` will not equal `this.currentSource_` and `previousSource` will be set to `this.currentSource_` before it is set to null)

Scenario 2:
1. setSource is called
2. `loadstart` happens (`previousSource` will be set to `this.currentSource_` for thes first time)
3. `loadstart` happens again (`previousSource` is still set to `this.currentSource_` do nothing